src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1998,4 +1998,40 @@
   <data name="UseConcreteTypeForParameterMessage" xml:space="preserve">
     <value>Change type of parameter '{0}' from '{1}' to '{2}' for improved performance</value>
   </data>
+  <data name="PreferLengthOverAnyCodeFixTitle" xml:space="preserve">
+    <value>Use 'Length' check instead of 'Any()'</value>
+  </data>
+  <data name="PreferLengthOverAnyTitle" xml:space="preserve">
+    <value>Prefer 'Length' check over 'Any()'</value>
+  </data>
+  <data name="PreferLengthOverAnyMessage" xml:space="preserve">
+    <value>Prefer comparing 'Length' to 0 rather than using 'Any()', both for clarity and for performance</value>
+  </data>
+  <data name="PreferLengthOverAnyDescription" xml:space="preserve">
+    <value>Prefer a 'Length' check over an 'Any()' call. The 'Any()' call will convert the collection into an 'IEnumerable' which creates the need for devirtualization at runtime.</value>
```

Comment:
```suggestion
    <value>Prefer comparing 'Length' to 0 rather than using 'Any()'. It is clearer and the 'Any()' call could be slower especially if it uses the 'IEnumerable.Any()' extension method</value>
```
The wording still could use more improvement, CC @gewarren @stephentoub